example/app/(examples)/DataBinding.tsx

```diff
@@ -1,23 +1,48 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import {
   SafeAreaView,
   ScrollView,
   StyleSheet,
   Button,
   View,
 } from 'react-native';
-import Rive, { Fit, RiveRef } from 'rive-react-native';
+import Rive, {
+  Fit,
+  RiveRef,
+  useRiveColor,
+  useRiveNumber,
+  useRiveReady,
+  useRiveString,
+} from 'rive-react-native';
 
 export default function DataBinding() {
   const riveRef = React.useRef<RiveRef>(null);
+  const riveIsReady = useRiveReady(riveRef);
 
+  let [buttonText, setButtonText] = useRiveString(riveRef, 'Button/State_1');
+  let [lives, setLives] = useRiveNumber(riveRef, 'Energy_Bar/Lives');
+  let [barColor, setBarColor] = useRiveColor(riveRef, 'Energy_Bar/Bar_Color');
+
+  useEffect(() => {
+    if (riveIsReady) {
+      // Set initial values through hooks
+      setButtonText("Let's go!");
+      setLives(7);
+      setBarColor({ r: 0, g: 255, b: 0, a: 255 });
+    }
+  }, [riveIsReady, setButtonText, setLives, setBarColor]);
+
+  console.log('Button Text:', buttonText);
+  console.log('Lives:', lives);
+  console.log('Bar Color:', barColor);
```

Comment:
These only every log `undefined`. I'd say either delete them or put them inside `riveIsReady`.

Reply:
I think you tested on iOS, which doesn't have the listener functionality yet on the native side! So it wouldn't have worked. These will log out the correct values as they get set from undefined to whatever type.